modules/oe_newsroom_newsletter/src/Plugin/Block/NewsletterSubscriptionBlock.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_newsroom_newsletter\Plugin\Block;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Form\FormBuilderInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\multivalue_form_element\Element\MultiValue;
+use Drupal\oe_newsroom_newsletter\Form\SubscribeForm;
+use Drupal\oe_newsroom_newsletter\OeNewsroomNewsletter;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a Newsletter Subscription Block.
+ *
+ * @Block(
+ *   id = "oe_newsroom_newsletter_subscription_block",
+ *   admin_label = @Translation("Newsletter Subscription Block"),
+ *   category = @Translation("OE Newsroom Newsletter")
+ * )
+ */
+class NewsletterSubscriptionBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The form builder.
+   *
+   * @var \Drupal\Core\Form\FormBuilder
+   */
+  private $formBuilder;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, FormBuilderInterface $form_builder) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->formBuilder = $form_builder;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('form_builder')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state) {
+    $form = parent::blockForm($form, $form_state);
+
+    $config = $this->getConfiguration();
+
+    $form['newsletters_language'] = [
+      '#type' => 'language_select',
+      '#title' => $this->t('Select the selectable languages for newsletter'),
+      '#description' => $this->t('Empty = all possible languages. For unilingual distribution lists select the correct language. If one language is selected, this will remain hidden from the user on the (un)subscribe from. Only site languages can be chosen as the content is normally pulled from the site.'),
```

Comment:
No need for this long explanation.
```suggestion
      '#description' => $this->t('Select the languages in which the newsletter is available. Leave empty to show all languages. Only the languages currently enabled in the site are available.'),
```